### ISSUE TITLE
Flow.monitor signature is not nice, would be monitorMat 

### ISSUE THREAD

**Issue by ktoso** [akka/akka#24812]
Likely we can't fix it, but monitor signature is now:

`  def monitor[Mat2]()(combine: (Mat, FlowMonitor[Out]) ⇒ Mat2): ReprMat[Out, Mat2] =`

which is a "Mat" style. 

It should be:

`  def monitorMat[Mat2]()(combine: (Mat, FlowMonitor[Out]) ⇒ Mat2): ReprMat[Out, Mat2] =`

and 

`  def monitor(): ReprMat[FlowMonitor[Out]] =`

Though I don't think we can fix it nowadays due to bin compat

**Comment by johanandren** [comment]
I agree, would have been a bit nicer.

**Comment by ktoso** [comment]
I was worried if the new signature would conflict with the existing one though due the multiple param list hm... maybe it'd work

**Comment by ktoso** [comment]
> Add, deprecate, remove (later)

Yeah so sadly it is as I feared and it does cause source incompatibility in Scala, Did a quick try about this
